trino-s3-proxy/src/main/java/io/trino/s3/proxy/server/credentials/SigningController.java

```diff
@@ -36,17 +37,56 @@ public SigningController(CredentialsController credentialsController, SigningCon
         maxClockDrift = signingControllerConfig.getMaxClockDrift().toJavaTime();
     }
 
+    public Optional<SigningMetadata> signingMetadataFromRequest(
+            Function<Credentials, Credentials.Credential> credentialsSupplier,
+            URI requestURI,
+            MultivaluedMap<String, String> requestHeaders,
+            MultivaluedMap<String, String> queryParameters,
+            String httpMethod,
+            String encodedPath)
+    {
+        String authorization = requestHeaders.getFirst("Authorization");
+        if (authorization == null) {
+            return Optional.empty();
+        }
+
+        List<String> authorizationParts = Splitter.on(",").trimResults().splitToList(authorization);
```

Comment:
Should we treat these cases separately? It seems that S3 may return different error codes depending on what failed with authorization: https://docs.aws.amazon.com/AmazonS3/latest/API/ErrorResponses.html

I'm wondering whether we should distinguish between no authorization being present, an invalid authorization being provided (e.g., an incorrect secret key) and entirely malformed authorization (e.g., multiple authorization headers)

Reply:
I think we need a controller of some kind to handle error responses so that we can ensure we return the right codes. We could add that now or later. wdyt?

Reply:
I think we can focus on getting this working as an MVP and then follow up later with these tweaks - opened #18 for that